django/db/models/fields/related.py

```diff
@@ -778,9 +778,7 @@ def get_attname_column(self):
 
     def get_joining_columns(self, reverse_join=False):
```

Comment:
The signature of this method is changing.
Should this be noted in a release note?

Reply:
Since you are returning `Field` instances now I think it might be worth having a new method instead that returns _compilable_ expression tuples instead and could possibly take care of the casting itself so `Join` remains agnostic of that.

Then `Join` could then raise a deprecation warning if the new method is missing can keep calling `get_joining_columns` until the deprecation period ends.

Reply:
Makes sense 👍  I pushed some edits :)